### ISSUE TITLE
Architect handling of show durations

### ISSUE THREAD

**Issue by jensimmons** [jensimmons/thewebahead#30]
I'm having a hard time figuring out what's the best way to handle the data regarding the length of each episode. Here's the deal:

Every episode has a length — a number of hours, minutes, and seconds of the show. Such times are usually formatted as 01:26:13 or 01h 26m 13s. 

I want to create a field that holds this data.

I also want to be able to make a calculation that adds up all of the show durations, and tells me the total number of hours, minutes, and seconds. Example: 88 shows = 179 hours, 23 minutes, and 12 seconds (I totally just made up that number. But I'd like to know what the real number is.)

I might someday want to be able to graph this data, and look at trend charts. Are the shows getting longer? Shorter? Etc. I have no specific plans, but if the data is stored well, this becomes more possible in case someday I do have a need for it. I hate having badly-stored data. 

So...

I started with a simple text field, and currently save things as "XX:YY:ZZ", but that's dumb data. No math is possible. It works fine for printing the number back out onto a page, but it's useless for anything else, and prone to data entry errors.

I attempted to create a data module field, collecting only hour, minute and seconds (no month or day), but date module requires a year. I don't believe you can not collect a year for every "date". Which I could fudge and just, whatever, collect a fake year... but I'm guessing that bodes poorly for being able to add the durations or do anything else with the data. Might as well just use the text field.

I tried installing Time Field (https://www.drupal.org/project/timefield), but I couldn't get it to print out the timestamp. It wants everything to be a time like "2:15pm". The description of the module makes it seem like this is the solution, but I couldn't figure it out. It seems under-supported / unfinished. And I'm not sure how I could accomplish the desired addition. Even with Views Calc. (Is Views Calc going to understand a timefield field value and add them?)

I'm tempted to create three text fields (or better, integer fields) of 2 characters each: 
[ __ __ ]  [ __ __ ]  [ __ __ ] 
and then enter hh mm ss into those three fields. I could add each in Views Calc, getting a very large number of seconds, a very large number of minutes, and very large number of hours. Manually divide seconds and minutes by 60, carrying the totals over into the next column, and creating a total time of all shows. It'd be a pain to have to manually create a useful number, but at least the hard work of adding 88 columns of numbers would be done by robots. The downside of this approach: it's a lot of work to enter all the data into three separate fields, knowing that it's a hacky solution, and knowing that if something better came along, all the data would have to be reentered. 

There's no way I'm the only person on the planet who's needed time duration fields in Drupal, with the ability to do something useful with the data. (Like hey, sort by length of show!) I'd be surprised if there isn't already a module stack to do this. I just don't know what that is. Am I just missing something about Date?

What's the best way to do this?

**Comment by jensimmons** [comment]
Jibran Ijaz suggested Interval module on twitter: https://twitter.com/JibranIjaz/status/534870994817941504

https://www.drupal.org/project/interval

**Comment by larowlan** [comment]
There is also https://www.drupal.org/project/timeperiod, but I think interval is more widely used, its a requirement for commerce recurring and a few others.

There's a D8 port but its a few months out of date

**Comment by chipcullen** [comment]
What is your source for this time data? Are you simply looking at a player that shows --h --m --s? 

I only ask, because, if your concern is keeping the data be future-friendly, could you not simply just enter the seconds of an episode? (e.g. a 5 minute episode would be 300 seconds) That would also be the simplest thing to do over the course of 80-odd episodes.

Then you could have your robots do the transforming of the raw second count into the display of your choice. [Looks like it's not that hard, actually](http://stackoverflow.com/questions/3172332/convert-seconds-to-hourminutesecond).

This is all predicated on the idea, though, that the total amount of seconds is easy to glean from whatever media source you have.

**Comment by larowlan** [comment]
timeperiod stores it in seconds, but lets you enter in h m s etc

On 19 November 2014 12:59, Chip Cullen notifications@github.com wrote:

> What is your source for this time data? Are you simply looking at a player
> that shows --h --m --s?
> 
> I only ask, because, if your concern is keeping the data be
> future-friendly, could you not simply just enter the seconds of an episode?
> (e.g. a 5 minute episode would be 300 seconds) That would also be the
> simplest thing to do over the course of 80-odd episodes.
> 
> Then you could have your robots do the transforming of the raw second
> count into the display of your choice. Looks like it's not that hard,
> actually
> http://stackoverflow.com/questions/3172332/convert-seconds-to-hourminutesecond
> .
> 
> This is all predicated on the idea, though, that the total amount of
> seconds is easy to glean from whatever media source you have.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jensimmons/thewebahead/issues/30#issuecomment-63585077
> .

## 

Lee Rowlands
Senior Drupal Developer
PreviousNext
skype: larowlan
email: larowlan@previousnext.com.au

**Comment by jensimmons** [comment]
Time Period is the solution! That creates exactly the kind of field I want, simply. Thanks @larowlan! Even will all the searching I was doing, I hadn't found that one. 

I've now got a new field, an integer field, and am converting over the data. I've built a table, with Views Calc, and it gives me the grand total number of seconds. I'll probably just convert the seconds into hours, mins, seconds manually. Maybe someday a custom module can do the last bit.
